pkg/identity/identity.go

```diff
@@ -98,3 +101,20 @@ func (id *Identity) GetModel() *models.Identity {
 func NewIdentity(id NumericIdentity, lbls labels.Labels) *Identity {
 	return &Identity{ID: id, Labels: lbls}
 }
+
+// PrefixString returns the IPIdentityPair's IP as either a host IP in the
+// format w.x.y.z if 'host' is true, or as a prefix in the format the w.x.y.z/N
+// if 'host' is false.
+func (pair *IPIdentityPair) PrefixString(host bool) string {
```

Comment:
Yeah this is a bit ugly, I haven't spent any time trying to clean it up. You're right that (1) is a special case, though in theory this could be handled by setting the mask to nil. For (2), there are a couple of other special cases. The delete case is simple and we don't require the parameter there. The other case is in `findShadowedCIDR`, where it's possible to have both `192.0.2.3` and `192.0.2.3/32`. In this case, the identity for the host entry is prioritized. Both could exist in the kvstore at once, and when the host is deleted, then the ipcache pushes down an update to use the CIDR prefix's identity instead of the host's identity. In this case we need to be able to turn the host entry into a longest-possible-match prefix for the protocol in use (IPv4/IPv6). We could do a bit of special code for that case instead of having this parameter though.

Reply:
@raybejjani I added a new commit on top for this.